File: javascript/ch5/MaybeFactory.js
class MaybeFactory {
  static just(value) {
    const { Just } = require('./Just');
    return new Just(value);
  }

  static nothing() {
    const { Nothing } = require('./Nothing');
    return new Nothing();
  }

  static fromNullable(value) {
    return value != null ? MaybeFactory.just(value) : MaybeFactory.nothing();
  }

  static of(value) {
    return MaybeFactory.just(value);
  }
}

module.exports = {
  MaybeFactory
};

File: python/python/ch5/MaybeFactory.py
# from .Just import Just
# from .Nothing import Nothing

class MaybeFactory:
    @staticmethod
    def just(value):
        from .Just import Just
        return Just(value)
      
    @staticmethod
    def nothing():
        from .Nothing import Nothing
        return Nothing()
      
    @staticmethod
    def from_nullable(value):
        return MaybeFactory.just(value) if value != None else MaybeFactory.nothing()
      
    @staticmethod
    def of(value):
        return MaybeFactory.just(value)